test/apps/iodemo/io_demo.cc

```diff
@@ -1343,10 +1373,56 @@ class DemoClient : public P2pDemoCommon {
         }
     }
 
+    void active_servers_swap(size_t index1, size_t index2) {
+        size_t& active_server1 = _active_servers[index1];
+        size_t& active_server2 = _active_servers[index2];
+
+        std::swap(_server_info[active_server1].active_index,
+                  _server_info[active_server2].active_index);
+        std::swap(active_server1, active_server2);
+    }
+
+    void active_servers_insert(size_t server_index) {
+        assert(_num_active_servers_to_use <= _active_servers.size());
+
+        _active_servers.push_back(server_index);
+
+        size_t last_index = _active_servers.size() - 1;
+        _server_info[last_index].active_index = last_index;
+
+        active_servers_make_used(last_index);
+        assert(_num_active_servers_to_use <= _active_servers.size());
+    }
+
+    void active_servers_remove(size_t active_index) {
+        assert(active_index < _active_servers.size());
+
+        if (active_index < _num_active_servers_to_use) {
+            active_index = _num_active_servers_to_use - 1;
+            active_servers_make_unused(active_index);
```

Comment:
```
            active_servers_make_unused(active_index);
            active_index = _num_active_servers_to_use;
```
or even maybe return new (unused) index from `active_servers_make_unused` to make in clearer:
```
active_index = active_servers_make_unused(active_index);
```

Reply:
done 1st approach, since the 2nd requires the same update in `active_servers_make_used()` to align them